build.ps1

```diff
@@ -868,11 +870,21 @@ $items.GetEnumerator() | %{
 	$item.Name = $name
 	$item.ArchiveFile = $archiveFile
 	$item.PatchDirectory = $(New-Object System.IO.DirectoryInfo $patchDirectory)
-	$item.BuildDirectory = $(New-Object System.IO.DirectoryInfo "$workingDirectory\$($archiveFile.BaseName)")
-	$item.BuildArchiveFile = $(New-Object System.IO.FileInfo "$workingDirectory\$($archiveFile.BaseName)-$filenameArch$($archiveFile.Extension)")
 	$item.Dependencies = @($item.Dependencies | %{ $items[$_] })
 	$item.Dependents = @()
 	$item.State = ''
+
+	switch ([System.IO.Path]::GetExtension($item.ArchiveFile)) {
+		'.xz' {
+			$filename = [System.IO.Path]::GetFileNameWithoutExtension($item.ArchiveFile)
+			$tarFile = New-Object System.IO.FileInfo "$workingDirectory\$filename"
```

Comment:
Here and the next line is where you *should* use Path.GetFileNameWithoutExtension

```powershell
$item.BuildDirectory = $(New-Object System.IO.DirectoryInfo "$workingDirectory\$([System.IO.Path]::GetFileNameWithoutExtension($archiveFile))")
```